File: lutech_erp_next/public/js/customer.js
// Copyright (c) 2026, Lutech and contributors
// License: MIT

frappe.ui.form.on("Customer", {
	refresh(frm) {
		set_customer_name_from_parts(frm);
	},
	custom_prenom(frm) {
		set_customer_name_from_parts(frm);
	},
	custom_nom_famille(frm) {
		set_customer_name_from_parts(frm);
	},
});

function set_customer_name_from_parts(frm) {
	const full = [frm.doc.custom_prenom, frm.doc.custom_nom_famille]
		.filter(Boolean)
		.join(" ")
		.trim();
	if (full && frm.doc.customer_name !== full) {
		frm.set_value("customer_name", full);
	}
}

File: lutech_erp_next/overrides/customer.py
import frappe
from frappe import _


def validate(doc, method=None):
	"""Keep customer_name in sync with Prénom + Nom de Famille."""
	parts = [doc.get("custom_prenom"), doc.get("custom_nom_famille")]
	full = " ".join(p.strip() for p in parts if p and str(p).strip())
	if full:
		doc.customer_name = full
	elif not doc.customer_name:
		frappe.throw(_("Please set Prénom and Nom de Famille"))
